### ISSUE TITLE
Capability for comments

### ISSUE THREAD

**Issue by tobiasKaminsky** [nextcloud/server#20612]
- [ ] Android @tobiasKaminsky 
- [ ] ~~Desktop @camilasan~~
- [ ] iOS @marinofaggiana 
- [x] Server @nextcloud/server-triage 
- [ ] deck @stefan-niedermann @juliushaertl 

Comments can be disabled by `occ app:disable comments`
Therefore we need a capability that allows clients to react on it in a proper way

**Comment by stefan-niedermann** [comment]
If this has been implemented in the `capabilities`-endpoint, it would be very nice if the one who implemented this could post a sample `JSON`-response with enabled and disabled capabilities here (would make the implementation way easier, at least for us :wink:).

**Comment by rullzer** [comment]
No comments on desktop ;)

**Comment by tobiasKaminsky** [comment]
> No comments on desktop ;)

not yet :-D

**Comment by blizzz** [comment]
> Comments can be disabled by occ app:disable comments

It disables only the web UI, but the functionality and the API is there itself. Still, it is reasonable to indicate that file comments ain't offered anywhere.

**Comment by stefan-niedermann** [comment]
To be honest i don't see the point of "disabling" the comments app if it is still possible to post comments via API...

What's the use case? Either i don't want people to comment, then i should deny it on all levels, or i don't care at all.

**Comment by szaimen** [comment]
> To be honest i don't see the point of "disabling" the comments app if it is still possible to post comments via API...

The biggest point of disabling the comments app imo is, that you get a simpler web interface (the sidebar is already really overpopulated with tabs) and e.g. you have talk for chatting and comments for chatting. 

Since talk enables videochat on files and is more useful imo, I have disabled the comments app and only allow talk on the webinterface...

**Comment by stefan-niedermann** [comment]
But can't this lead to missed comments which might be posted by a 3rd-party-app via API? I get the point, but without actually *disabling* and *denying* comments this might be dangerous as more and more integrations-apps are coming 😉

**Comment by szaimen** [comment]
> But can't this lead to missed comments which might be posted by a 3rd-party-app via API? I get the point, but without actually _disabling_ and _denying_ comments this might be dangerous as more and more integrations-apps are coming

Potentially yes, and I actually wouldn't disable the comments app if it was well integrated into the activities tab like it was already suggested by jancborchardt elsewhere and as it already is in the Nextcloud Android app.

**Comment by stefan-niedermann** [comment]
cc @jancborchardt and @nextcloud/designers

**Comment by nickvergessen** [comment]
No, when the comments app is disabled the API does not allow to comment on files. But e.g. you can still comment on announcements, deck, and others.

This issue here was opened, so the mobile clients for the files don't show the comments tab when you can't comment on files.

**Comment by szaimen** [comment]
Reopen for client implementation?

**Comment by nickvergessen** [comment]
Just for the record as Deck is in the list. The capability is only for file comments, so I guess it does not affect deck, right?

**Comment by nickvergessen** [comment]
No, it registers it's own entity type, so deck comments can be made without the comments app:
https://github.com/nextcloud/deck/blob/master/lib/AppInfo/Application.php#L152-L164

**Comment by stefan-niedermann** [comment]
@nickvergessen i referred to this PR in the deck app repo: https://github.com/nextcloud/deck/pull/1734/files

But according to your comment the Deck app needs to provide a own REST API then to expose the enabled / disabled status of its comments then?

Hmm @juliushaertl can you clearify this please?

**Comment by stefan-niedermann** [comment]
> we can actually just always keep them available in deck.

@juliushaertl can you please give me an update whether or not you will keep them always available in deck? In this case this issue is obsolete for the deck android app.

**Comment by juliushaertl** [comment]
Since the Api is always enabled I think it makes most sense to just always have comments enabled in deck then. 

https://github.com/nextcloud/deck/issues/2126
